### PR TITLE
Add callback for ffmpeg process exit

### PR DESCRIPTION
**Pull request by normen**
* Report error when ffmpeg fails
* Use SIGTERM to quit ffmpeg

### CODE REVIEW

**Comment by normen** [comment]
Is there something we could signal to HomeKit when ffmpeg fails? If you have a hint I could try and add that too.

**Comment by KhaosT** [comment]
You can call `StreamController.forceStop()` in that case to inform about status change. The basic way to do that is iterate through `self.streamControllers`, find the one with matching `sessionID` and invoke `forceStop` on it.

**Comment by normen** [comment]
It doesn't seem like my iPhone is showing the "Camera not available" any sooner with this change but I believe you when you say its the thing to do :)

File: ffmpeg.js [hunk 1]
```diff
@@ -331,6 +331,12 @@ FFMPEG.prototype.handleStreamRequest = function(request) {
             this.log("Stopped streaming");
           } else {
             this.log("ERROR: FFmpeg exited with code " + code);
+            for(var i=0; i < this.streamControllers.length; i++){
+              var controller = this.streamControllers[i];
+              if(controller.sessionIdentifier != null && sessionID.equals(controller.sessionIdentifier)){
```
Comment (KhaosT) [inline ffmpeg.js]: why not using `===`?

**Comment by KhaosT** [comment]
Ah that won't mark the camera as unavailable, it only inform iOS about the streaming session got terminated...

File: ffmpeg.js (hunk 1 above)
Comment (normen) [reply ffmpeg.js]: Cause I'm a JS noob I guess :)

**Comment by normen** [comment]
Yeah I thought thats what iOS does then. Also happens when the stream fails otherwise. Whatever, it was mainly about closing properly in terms of HAP-NodeJS.

**Comment by normen** [comment]
Oh wait there was two things in the README too..

File: ffmpeg.js [hunk 2]
```diff
@@ -326,17 +326,28 @@ FFMPEG.prototype.handleStreamRequest = function(request) {
...
+          } else {
+            this.log("ERROR: FFmpeg exited with code " + code);
+            for(var i=0; i < this.streamControllers.length; i++){
+              var controller = this.streamControllers[i];
+              if(controller.sessionIdentifier === sessionID){
+                controller.forceStop();
+              }
+            }
+          }
+        });
```
Comment (KhaosT) [inline ffmpeg.js]: If you are going to use `this` in a block, you need to `.bind(this)` on the block.

Comment (normen) [reply ffmpeg.js]: Oh.. or just use a self yeah, didn't realize I was in an anonymous inner one there. Guess it picked up the log from the Platform then..

File: ffmpeg.js [hunk 3]
```diff
@@ -326,13 +326,14 @@ FFMPEG.prototype.handleStreamRequest = function(request) {
             console.log(data.toString());
           });
         }
+        var self = this;
```
Comment (KhaosT) [inline ffmpeg.js]: Just a minor thing, can we use `let self` instead of `var self`?

**Comment by KhaosT** [comment]
Thanks :)

**Comment by normen** [comment]
Any time, thank you for helping. And for my home automation in general ;)

**Pull request merged**